spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,454 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Authentication::AuthnAzure::ValidateApplicationIdentity' do
+  let(:account) { "account" }
+  let(:service_id) { "serviceId" }
+  let(:username) { "host/azureTestVM" }
+  let(:hostname) { "azureTestVM" }
+
+  let(:resource_class) { double("ResourceClass") }
+  let(:mocked_host) { double "MockHost" }
```

Comment:
The value inside `double` should be the class you mock. For example, the double in `resource_class` should be `Resource`.
Also, add brackets in the `double` constructor.

Reply:
Also - please choose one approach between adding `mocked_` and not adding the prefix to your mocks.

Reply:
Your first point, changing
Your second point, I didn't understand about adding brackets? I added the following if this is what you meant:
```
    allow(resource_class).to receive(:[])
                               .with("#{account}:host:#{hostname}")
                               .and_return(mocked_host)
```

Reply:
what i meant is to change `double "MockHost"` to `double("MockHost")`